velox/exec/tests/OperatorTraceTest.cpp

```diff
@@ -620,4 +625,167 @@ TEST_F(OperatorTraceTest, traceTableWriter) {
+  const uint32_t crc32 = folly::crc32(
+      reinterpret_cast<const uint8_t*>(split.data()), split.size());
+  const auto splitInfoFile = fs->openFileForWrite(
+      fmt::format(
+          "{}/{}", traceDir->getPath(), OperatorTraceTraits::kSplitFileName),
+      filesystems::FileOptions{.shouldThrowOnFileAlreadyExists = false});
+  auto ioBuf = folly::IOBuf::create(12);
+  folly::io::Appender appender(ioBuf.get(), 0);
+  appender.writeLE(length);
+  appender.push(reinterpret_cast<const uint8_t*>(split.data()), length);
+  splitInfoFile->append(std::move(ioBuf));
+  splitInfoFile->close();
+
+  const auto reader = exec::trace::OperatorTraceSplitReader(
+      {traceDir->getPath()}, memory::MemoryManager::getInstance()->tracePool());
+  auto actualSplits = reader.read();
+  for (int i = 0; i < numSplits; ++i) {
+    ASSERT_FALSE(actualSplits[i].hasGroup());
+    ASSERT_TRUE(actualSplits[i].hasConnectorSplit());
+    const auto actualConnectorSplit = actualSplits[i].connectorSplit;
+    const auto expectedConnectorSplit = splits[i].connectorSplit;
+    ASSERT_EQ(
+        actualConnectorSplit->toString(), expectedConnectorSplit->toString());
+  }
+}
+
+TEST_F(OperatorTraceTest, traceSplitCorrupted) {
+  constexpr auto numSplits = 3;
+  std::vector<exec::Split> splits;
+  for (int i = 0; i < numSplits; ++i) {
+    auto builder = HiveConnectorSplitBuilder(fmt::format("path-{}-{}", i, i));
+    const auto key = fmt::format("k{}", i);
+    const auto value = fmt::format("v{}", i);
+    splits.emplace_back(
+        builder.start(i)
+            .length(i)
+            .connectorId(fmt::format("{}", i))
+            .fileFormat(dwio::common::FileFormat(i + 1))
+            .infoColumn(key, value)
+            .partitionKey(
+                key, i > 1 ? std::nullopt : std::optional<std::string>(value))
+            .tableBucketNumber(i)
+            .build(),
+        -1);
+  }
+
+  const auto traceDir = TempDirectoryPath::create();
+  auto writer = exec::trace::OperatorTraceSplitWriter(traceDir->getPath());
+  for (int i = 0; i < numSplits; ++i) {
+    writer.write(splits.at(i));
+  }
+  writer.finish();
+
+  // Append a split with wrong checksum to the split info file.
+  const auto fs = filesystems::getFileSystem(traceDir->getPath(), nullptr);
+  const std::string split = "deadbeaf";
+  const uint32_t length = split.length();
+  const uint32_t crc32 = folly::crc32(
+      reinterpret_cast<const uint8_t*>(split.data()), split.size());
+  const auto splitInfoFile = fs->openFileForWrite(
+      fmt::format(
+          "{}/{}", traceDir->getPath(), OperatorTraceTraits::kSplitFileName),
+      filesystems::FileOptions{.shouldThrowOnFileAlreadyExists = false});
+  auto ioBuf = folly::IOBuf::create(16);
+  folly::io::Appender appender(ioBuf.get(), 0);
+  appender.writeLE(length);
+  appender.push(reinterpret_cast<const uint8_t*>(split.data()), length);
+  appender.writeLE(crc32 - 1);
+  splitInfoFile->append(std::move(ioBuf));
+  splitInfoFile->close();
```

Comment:
You could add one more valid split after this to see if the trace split reader stops at the first corrupted split. Thanks!